File: state/java/OrderFood.java
package state.java;

public class OrderFood {
    private OrderPhase orderPhase;

    public OrderFood(OrderPhase orderPhase) {
        this.orderPhase = orderPhase;
        this.switchOrderPhase(orderPhase);
    }

    public void switchOrderPhase(OrderPhase orderPhase) {
        this.orderPhase.setOrderPhase(this);
    }

    public void startOrder() {
        this.orderPhase.startOrder();
    }

    public void endOrder() {
        this.orderPhase.endOrder();
    }

    public void deliverOrder() {
        this.orderPhase.deliverOrder();
    }
}

File: state/golang/order_food.go
package main

// OrderFood is the Context
type OrderFood struct {
	State OrderPhase
}

func (o *OrderFood) SetState(state OrderPhase) {
	o.State = state
}

func (o *OrderFood) AddItem() {
	o.State.AddItem()
}

func (o *OrderFood) Confirm() {
	o.State.Confirm()
}
